src/api/entities/SecurityToken/TokenHolders.ts

```diff
@@ -27,16 +42,56 @@ export class TokenHolders extends Namespace<SecurityToken> {
 
     const { entries, lastKey: next } = await requestPaginated(query.asset.balanceOf, {
       arg: rawTicker,
-      paginationOpts,
+      paginationOpts: opts?.pagination || undefined,
     });
 
-    const data: IdentityBalance[] = entries.map(([storageKey, balance]) => ({
-      identity: new Identity(
-        { did: identityIdToString(storageKey.args[1] as IdentityId) },
-        context
-      ),
-      balance: balanceToBigNumber(balance),
-    }));
+    let data: IdentityBalance[] = [];
+
+    const securityToken = new SecurityToken({ ticker }, context);
+
+    const areFrozen = opts?.mintStatus ? await securityToken.transfers.areFrozen() : undefined;
+
+    if (opts?.mintStatus && !areFrozen) {
+      const entriesChunks = chunk(entries, 10);
+
+      await Promise.all(
+        entriesChunks.map(async entriesChunk => {
+          const canMint = await Promise.all(
+            entriesChunk.map(([storageKey]) => {
+              return securityToken.transfers.canMint({
+                to: identityIdToString(storageKey.args[1] as IdentityId),
+                amount: new BigNumber(1),
+              });
+            })
+          );
+
+          entriesChunk.forEach(([storageKey, balance], index) => {
+            data.push({
+              identity: new Identity(
+                { did: identityIdToString(storageKey.args[1] as IdentityId) },
+                context
+              ),
+              balance: balanceToBigNumber(balance),
```

Comment:
All this should be only done once outside the if/else and then you can chunk `data` and process that instead